benchmark/test_biocypher_query_generation.py

```diff
@@ -13,142 +19,284 @@
 
 MODEL_NAMES = [
     "gpt-3.5-turbo",
-    "gpt-4",
+    # "gpt-4",
 ]
 
 
 @pytest.fixture(scope="module", params=MODEL_NAMES)
 def prompt_engine(request):
-    model_name = request.param
-    return BioCypherPromptEngine(
-        schema_config_or_info_path="test/test_schema_info.yaml",
-        model_name=model_name,
+    def setup_prompt_engine(kg_schema_path):
+        model_name = request.param
+        return BioCypherPromptEngine(
+            schema_config_or_info_path=kg_schema_path,
+            model_name=model_name,
+        )
+
+    return setup_prompt_engine
+
+
+def benchmark_already_executed(
+    task: str,
+    subtask: str,
+    model_name: str,
+    result_files: dict[str, pd.DataFrame],
+) -> bool:
+    """
+    Checks if the benchmark task and subtask for the model_name have already
+    been executed.
+
+    Args:
+        task (str): The benchmark task, e.g. "biocypher_query_generation"
+        subtask (str): The benchmark subtask, e.g. "entities"
+        model_name (str): The model name, e.g. "gpt-3.5-turbo"
+        result_files (dict[str, pd.DataFrame]): The result files
+
+    Returns:
+
+        bool: True if the benchmark task and subtask for the model_name has
+            already been run, False otherwise
+    """
+    task_results = result_files[f"benchmark/results/{task}.csv"]
+    task_results_subset = (task_results["model"] == model_name) & (
```

Comment:
For me this fails because the key "model" is not in the DF.

Reply:
Now fixed